### ISSUE TITLE
mypy should give error at the timing when the type alias set multiple `TypeVarTuple`-based type parameters is defined but not used

### ISSUE THREAD

**Issue by hyperkai** [python/mypy#20544]
*Memo:
- mypy test.py
- mypy 1.19.1
- Python 3.14.0

Defining the type alias set multiple [TypeVarTuple](https://docs.python.org/3/library/typing.html#typing.TypeVarTuple)-based type parameters, mypy doesn't give error while mypy gives the error, when using the type alias as shown below so mypy should give error at the timing when the type alias set multiple `TypeVarTuple`-based type parameters is defined but not used:

*Memo:
- pyright and pyrefly give error at the timing when the type alias set multiple `TypeVarTuple`-based type parameters is defined.

```python
''' New syntax '''
type TA[*Ts1, *Ts2] = tuple[*Ts1] | tuple[*Ts2] # No error
''' New syntax '''

''' Old syntax '''
# from typing import TypeAlias, TypeVarTuple

# Ts1 = TypeVarTuple('Ts1')
# Ts2 = TypeVarTuple('Ts2')

# TA: TypeAlias = tuple[*Ts1] | tuple[*Ts2] # No error
''' Old syntax '''

v: TA[*tuple[int, ...], *tuple[str, ...]] # Error
```

> error: More than one variadic Unpack in a type is not allowed

**Comment by aviralgarg05** [comment]
I would like to solve this issue, please assign me this issue

**Comment by A5rocks** [comment]
Feel free to fix things even if you aren't assigned!
